abstract_commands.xml

```diff
@@ -5,8 +5,26 @@
         is accessible, then \Fsize matching the register's actual size must be
         supported.
 
+        \begin{steps}{Perform the following sequence of operations:}
+        \item Halt the hart, if \Fprehalt is set.
+        \item Execute the Program Buffer, if \Fpreexec is set.
+        \item Copy data from the specified register into {\tt data}, if \Fwrite is clear.
+        \item Copy data from {\tt data} into the specified register, if \Fwrite is set.
+        \item Execute the Program Buffer, if \Fpostexec is set.
+        \item Resume the hart, if \Fpostresume is set.
+        \end{steps}
+
         <field name="0" bits="31:24" />
-        <field name="0" bits="23:22" />
+        <field name="prehalt" bits="23">
+            When 1, halt the hart before performing the rest of the command.
```

Comment:
Ok. For some reason I thought we had this in the previous 0.11 implementation, but if not, simplicity is good. It seems the debugger could basically do something like this by using normal breakpoint exceptions and normal code, which is then basically the debug monitor situation. 

I need to think about whether this is complicated. I keep thinking that this command state machine is really complicated, but then I realize that it is basically what our ROM code was doing before, and all of it can be done in code vs. hardware (as desired) and I get less freaked out.